### DISCUSSION TITLE
failed to launch CarlaUE4.sh

### DISCUSSION THREAD

**Discussion by renjieDLUT** [carla-simulator/carla#4359] [Q&A]
```shell
$ ./CarlaUE4.sh
4.24.3-0+++UE4+Release-4.24 518 0
Disabling core dumps.
INTEL-MESA: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0

X Error of failed request:  BadDrawable (invalid Pixmap or Window parameter)
  Major opcode of failed request:  149 ()
  Minor opcode of failed request:  4
  Resource id in failed request:  0x2000041
  Serial number of failed request:  376
  Current serial number in output stream:  387
terminating with uncaught exception of type std::__1::system_error: mutex lock failed: Invalid argument
Signal 6 caught.
Segmentation fault (core dumped)
```

**Comment by renjieDLUT** [comment]
Have solved  the problem. Vulkan is the default graphics API, in Linux, and it may lead to some issues. selecting opengl as carla's graphic  API solve the problem perfectly.
`$ ./CarlaUE4.sh -opengl`

**Reply by qhaas** [reply]
A few caveats with [OpenGL vs Vulkan](https://carla.readthedocs.io/en/0.9.11/adv_rendering_options/#vulkan-vs-opengl).  OpenGL has artifacts under certain weather conditions (Issue #3377).  While your particular issue with Vulkan looks to involve some Intel related driver issues, know that headless (as in xorg-less) [does not work with vulkan without an upgrade to UE 4.25+](https://github.com/carla-simulator/carla/issues/3800#issuecomment-768993043)

**Reply by MalikSumbal** [reply]
I am new to Carla Simulator. I would be thankful if you could answer the following questions:

Does Carla work best on Windows or Linux?
Can we run the Carla simulator without GPU?
Is it possible to install and run Carla on Google Cloud?

**Reply by corkyw10** [reply]
@CarlaaUser 
CARLA is designed for Linux first but works on both OS's
See [here](https://carla.readthedocs.io/en/0.9.11/adv_rendering_options/#no-rendering-mode) for rendering options
If the VM is a compatible OS and you have sufficient resources, you should be able to run on GCP.

**Reply by MalikSumbal** [reply]
@corkyw10 Thank you.

**Reply by mutallibshaik** [reply]
Hi All, 

I installed Carla on an Ubuntu machine, but running the .CarlaUE4.sh does nothing. Can anybody please help?

**Reply by SaraMo14** [reply]
hi, did you solve the problem?

**Reply by JessicaYangyx** [reply]
> Hi All,
> 
> I installed Carla on an Ubuntu machine, but running the .CarlaUE4.sh does nothing. Can anybody please help?

I also meet with this problem. Has anybody solved it?

**Comment by dhdiagram4011** [comment]
hello!

(carla) simulator@simulator:~/apollo/carla_0.9.13$ ./CarlaUE4.sh
4.26.2-0+++UE4+Release-4.26 522 0
Disabling core dumps.
LowLevelFatalError [File:Unknown] [Line: 803] 
VulkanRHI::vkGetPhysicalDeviceSurfaceFormatsKHR(Device.GetPhysicalHandle(), Surface, &NumFormats, nullptr) failed, VkResult=-13
 at /home/jenkins/UnrealEngine_4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanSwapChain.cpp:179 
 with error 
Signal 11 caught.
Malloc Size=65538 LargeMemoryPoolOffset=65554 
CommonUnixCrashHandler: Signal=11
Malloc Size=131160 LargeMemoryPoolOffset=196744 
Malloc Size=131160 LargeMemoryPoolOffset=327928 
Engine crash handling finished; re-raising signal 11 for the default handler. Good bye.
Segmentation fault (core dumped)

help me!
Thank you

**Reply by Petilia** [reply]
> hello!
> 
> (carla) simulator@simulator:~/apollo/carla_0.9.13$ ./CarlaUE4.sh 4.26.2-0+++UE4+Release-4.26 522 0 Disabling core dumps. LowLevelFatalError [File:Unknown] [Line: 803] VulkanRHI::vkGetPhysicalDeviceSurfaceFormatsKHR(Device.GetPhysicalHandle(), Surface, &NumFormats, nullptr) failed, VkResult=-13 at /home/jenkins/UnrealEngine_4.26/Engine/Source/Runtime/VulkanRHI/Private/VulkanSwapChain.cpp:179 with error Signal 11 caught. Malloc Size=65538 LargeMemoryPoolOffset=65554 CommonUnixCrashHandler: Signal=11 Malloc Size=131160 LargeMemoryPoolOffset=196744 Malloc Size=131160 LargeMemoryPoolOffset=327928 Engine crash handling finished; re-raising signal 11 for the default handler. Good bye. Segmentation fault (core dumped)
> 
> help me! Thank you

Faced with the same problem

Can you tell me how to solve it?

**Comment by apcial1** [comment]
I'd like to seek guidance on how to install Carla on a server. I encountered an issue while attempting to install Carla version 0.9.10.1, where it stops immediately after running with the following message:

4.24.3-0+++UE4+Release-4.24 518 0 Disabling core dumps. (carla) carlauser@notebook-devenviron-0110-182655-1oignbn-notebook-0:/workspace/algorithm/carla$

Any assistance or advice on resolving this issue would be greatly appreciated.
